inference-engine/src/plugin_api/cpp_interfaces/interface/ie_iremote_context.hpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/**
+ * @brief This is a header file for the IE RemoteContext and RemoteBlob classes
+ *
+ * @file ie_remote_context.hpp
+ */
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "ie_parameter.hpp"
+#include "ie_remote_context.hpp"
+
+namespace InferenceEngine {
+
+class RemoteBlob;
+
+class INFERENCE_ENGINE_API_CLASS(IRemoteContext)
```

Comment:
Can we use internal `Impl` the same as we have for `ov::Core`?
It should allow to don't use `IRemoteContext` in public API.

Reply:
No, we can not. As we load `IRemoteContext` from the plugin we have to story its copy as shared_ptr and reference to `SO` object. The `Impl` would be just a struct with these two entities which is wrapped into another shared_ptr. 

I don't think that `IRemoteContext` confuse user more then `RemoteContext::Impl`. We can rename it to `RemoteContextImpl`. Or introduce namespace `ov::runtime::impl::RemoteContext`.